### PR TITLE
Pydantic for Test Scenario

### PR DESCRIPTION
**Pull request by amaslenn**
## Summary
Introduce Pydantic schema validation for Test Scenarios.

## Test Plan
1. CI

## Additional Notes
—

## Highlights for Release notes
We are working on schema improvements to simplify configs management and make them verifiable. This will help ensure that configs are correct before expensive runs on real hardware. Today we are enabling it for Test Scenario configs. This is a continuation of https://github.com/NVIDIA/cloudai/pull/145.

1. `Tests` becomes and array. This helps making case names more expressive:
    before:
    ```toml
    [Tests.1]
    # ...
    ```
    now:
    ```toml
    [[Tests]]
    id = "any-name.you_want" # before it was just "1"
    ```
2. `id` field is mandatory and must be unique and is used to specify dependencies:
    ```toml
    [[Tests]]
    id = "Tests.1"
    # ...

    [[Tests]]
    id = "Tests.2"
    # ...
      [[Tests.dependencies]]
      id = "Tests.1"
      # ...
    ```
3. `name` (under the list of tests) renamed to `test_name` to better reflect its meaning. It still references a test defined in a separate TOML file.
4. Dependencies converted to a list to support multiple dependencies of the same type.
    before
    ```toml
    # ...

    [Tests.2]
    name = "ucc_test_alltoall"
      [Tests.2.dependencies]
      start_post_comp = { name = "Tests.1", time = 0 }  # only one dependency of this type is allowed
    ```
    now
    ```toml
    # ...

    [[Tests]]
    id = "Tests.3"
    test_name = "ucc_test_alltoall"
    # ...
      [[Tests.dependencies]]
      type = "start_post_comp"
      id = "Tests.1"

      [[Tests.dependencies]]
      type = "start_post_comp"
      id = "Tests.2"
    ```

### CODE REVIEW

File: USER_GUIDE.md [hunk 1]
```diff
@@ -123,20 +123,24 @@ Test Scenario uses Test description from the previous step. Below is the `myconf
...
-
-[Tests.2]
-  name = "nccl_test_all_reduce_single_node"
-  time_limit = "00:20:00"
-  [Tests.2.dependencies]
-    start_post_comp = { name = "Tests.1", time = 0 }
+[[Tests]]
+id = "Tests.1"
+template_test = "nccl_test_all_reduce_single_node"
+time_limit = "00:20:00"
```
Comment (TaekyungHeo) [inline USER_GUIDE.md:129]: Let's discuss the new field names in the design meeting (id, template_test). 'ID' looks good to me, but I prefer 'test_definition' over 'template_test'.

**Comment by TaekyungHeo** [comment]
We should also have a corresponding PR in CloudAIX. The CloudAIX TOML files and code should be updated accordingly.

File: USER_GUIDE.md (hunk 1 above)
Comment (amaslenn) [reply USER_GUIDE.md:129]: Make sense, let's discuss.

Comment (amaslenn) [reply USER_GUIDE.md:129]: I gave `test_definition` a though and I think that it is misleading in this context: we reference a particular Test TOML which is based on a `TestDefinition`, but we do not reference definition directly.

Comment (TaekyungHeo) [reply USER_GUIDE.md:129]: This PR looks good to me, except for the field naming. If others are okay with the field name, I can approve.

Comment (srinivas212) [reply USER_GUIDE.md:129]: Why not test_name? We use 'name' in the test toml.

Comment (amaslenn) [reply USER_GUIDE.md:129]: `test_name` sounds good to me.
@TaekyungHeo @srivatsankrishnan what do you think?

Comment (TaekyungHeo) [reply USER_GUIDE.md:129]: `tests_name` looks good.

Comment (amaslenn) [reply USER_GUIDE.md:129]: Went ahead and renamed it here and in Cloud AI X as well.

Comment (srivatsankrishnan) [reply USER_GUIDE.md:129]: test_name looks good to me as well.

**Review by srivatsankrishnan — APPROVED** [review]
LGTM.

**Review by TaekyungHeo — APPROVED** [review]

**Review by TaekyungHeo — APPROVED** [review]

**Review by srivatsankrishnan — APPROVED** [review]

**Pull request merged**